lib/cucumber/rails.rb

```diff
@@ -13,7 +13,7 @@
   require 'cucumber/rails/action_dispatch'
   require 'rails/test_help'
 
-  unless Rails.application.config.cache_classes
+  unless Rails.application.config.cache_classes || defined?(Spring)
     warn "WARNING: You have set Rails' config.cache_classes to false
```

Comment:
... or are using Spring

Reply:
Ah, that becomes a more helpful message to the user, by including a note on Spring.

Reply:
I'm not sure what exactly you mean here, because the `cache_classes` settings is still explicitly set to `false` when Spring is used (Spring doesn't change this setting, it only implicitly requires it to be set to `false`).

Reply:
Oh, I see now that you've updated this. Well, as I said, I'm not sure "you have set `cache_classes` to false *or* you are using Spring" is accurate, because when you're using Spring the `cache_classes` setting still needs to be explicitly set to `false`. You *can* set it to `true`, and Spring won't complain about it, but then the code won't reload properly.

Reply:
Alternative formulations, to be accurate, yet still keep the suggestion that Spring could be the reason?

- (perhaps by using Spring)
- (running Spring needs cache_classes set to false)

Reply:
That's fine. Whatever you think is best I'm happy with.

Reply:
Running cucumber tests with spring and `config.cache_classes = true` in rails 6 fails with `reloading is disabled` error:
```
~/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.3.2/lib/active_support/dependencies/zeitwerk_integration.rb:14:in `rescue in block in clear':
reloading is disabled because config.cache_classes is true (RuntimeError)
```

The following warning message looks more clear for me (what do you think, guys?):
```text
WARNING: You have set Rails' config.cache_classes to false.
This is known to cause problems with database transactions.
Set config.cache_classes to true or run cucumber with Spring if you want to use transactions.
See https://github.com/jonleighton/spring-commands-cucumber on how to configure Spring for cucumber.
```

BTW the [original warning message](https://github.com/cucumber/cucumber-rails/blob/e897d6e012d85728cb3f06209dd5a5f3e2a01d01/lib/cucumber/rails3.rb#L15) was telling about issues with **using transactional fixtures** when `cache_classes` is set to `false`. And provides a link to the original issue (https://rspec.lighthouseapp.com/projects/16211/tickets/165) in a private project. In 5c7030a1 the warning message has been rephrased that disabling cache for classes causes problems with **database transactions**. Does it mean DatabaseCleaner's `:transaction` strategy under "database transactions"? Or transactional fixtures? Does this problem still persists? It would be good to copy the original issue from the private project to [cucumber-rails GH issues](https://github.com/cucumber/cucumber-rails/issues) and add link to it to the warning. What do you think @aslakhellesoy ?

Reply:
@Hirurg103 - When this PR was being formulated rails6 testing (On this repo), was still in its infancy, as such it's likely that this behaviour may have been modified by rails and our assumptions previously were correct but now may not be.

Also this error is coming from AS, a rails gem, which we don't control. We almost never throw an error for configuration in cucumber-rails, so you're probably best opening an issue on the rails/activesupport area.